docs/linking/2-request-consent-otp.puml

```diff
@@ -111,4 +119,76 @@ note over PISP, DFSP
   At this point, the PISP knows that the OTP authChannel is in use and the PISP App should prompt the user to provide the OTP.
 end note
 
+== Failure case ==
+
+autonumber 1 "<b>REQ-#</b>"
+
+PISP -> Switch ++: ""POST /consentRequests""\n\
+  ""  FSIOP-Source: pispa""\n\
+  ""  FSIOP-Destination: dfspa""\n\
+""{""\n\
+  ""  consentRequestId: "456",""\n\
+  ""  **userId: "username1234", //so the dfsp can associate this request with GET /accounts/**""\n\
+  ""  scopes: [ ""\n\
+  ""    { **accountId: "dfsp.username.1234", // Required for OTP!**""\n\
+  ""      actions: [ "accounts.transfer", "accounts.getBalance" ] },""\n\
+  ""    { accountId: "dfsp.username.5678",""\n\
+  ""      actions: [ "accounts.transfer", "accounts.getBalance" ] },""\n\
+  ""  ],""\n\
+  ""  authChannels: [ "Web", "OTP" ],""\n\
+  ""  callbackUri: "pisp-app://callback..."""\n\
+  ""}""
+Switch --> PISP: ""202 Accepted""
+deactivate PISP
+
+Switch -> DFSP ++:  ""POST /consentRequests""\n\
+  ""  FSIOP-Source: pispa""\n\
+  ""  FSIOP-Destination: dfspa""\n\
+""{""\n\
+  ""  consentRequestId: "456",""\n\
+  ""  userId: "username1234",""\n\
+  ""  scopes: [ ""\n\
+  ""    { accountId: "dfsp.username.1234",""\n\
+  ""      actions: [ "accounts.transfer", "accounts.getBalance" ] },""\n\
+  ""    { accountId: "dfsp.username.5678",""\n\
+  ""      actions: [ "accounts.transfer", "accounts.getBalance" ] },""\n\
+  ""  ],""\n\
+  ""  authChannels: [ "Web", "OTP" ],""\n\
+  ""  callbackUri: "pisp-app://callback..."""\n\
+  ""}""
+DFSP --> Switch: ""202 Accepted""
+deactivate Switch
+
+autonumber 1 "<b>DFSP-#</b>"
+
+DFSP -> DFSP: Verify the consentRequests is valid or not?
+
+note over PISP, DFSP
+  We respond with an error that the consentRequests is invalid so the PISP can inform the user.
+end note
+
+autonumber 5 "<b>REQ-#</b>"
+DFSP -> Switch ++: ""PUT /consentRequests/456/error""\n\
+  ""  FSIOP-Source: dfspa""\n\
+  ""  FSIOP-Destination: pispa""\n\
+  ""{""\n\
+  ""  errorInformation : { ""\n\
+  ""    errorCode: "7204", ""\n\
+  ""    errorDescription: "FSP does not support any requested scope actions" ""\n\
+  ""  } ""\n\
+  ""}""
+Switch --> DFSP: ""200 OK""
+deactivate DFSP
+
+Switch -> PISP ++: ""PUT /consentRequests/456/error""\n\
+  ""  FSIOP-Source: dfspa""\n\
+  ""  FSIOP-Destination: pispa""\n\
+  ""{""\n\
+  ""  errorInformation : { ""\n\
+  ""    errorCode: "7204", ""\n\
+  ""    errorDescription: "FSP does not support any requested scope actions" ""\n\
```

Comment:
NEW ERROR CASE: `authChannels` is an array that can have only one value. So we have an error case when specified channel isn't allowed for given userId

Reply:
for `POST /consentRequests` this must be an array with 2 values, so not really an issue... if the dfsp rejects the `POST /consentRequests`, that can be considered a generic rejection, right?